website/docs/r/stream_analytics_output_cosmosdb.html.markdown

```diff
@@ -0,0 +1,114 @@
+
+## Example Usage
+
+```hcl
+data "azurerm_resource_group" "example" {
+  name = "example-resources"
+}
+
+data "azurerm_stream_analytics_job" "example" {
+  name                = "example-job"
+  resource_group_name = azurerm_resource_group.example.name
+}
+
+resource "azurerm_cosmosdb_account" "example" {
+  name                = "exampledb"
+  resource_group_name = data.azurerm_resource_group.example.name
+  location            = data.azurerm_resource_group.example.location
+  offer_type          = "Standard"
+  kind                = "GlobalDocumentDB"
+
+  consistency_policy {
+    consistency_level       = "BoundedStaleness"
+    max_interval_in_seconds = 10
+    max_staleness_prefix    = 200
+  }
+
+  geo_location {
+    prefix            = "prefix-customid"
+    location          = azurerm_resource_group.example.location
+    failover_priority = 0
+  }
+}
+
+resource "azurerm_cosmosdb_sql_database" "example" {
+  name                = "cosmos-sql-db"
+  resource_group_name = azurerm_cosmosdb_account.example.resource_group_name
+  account_name        = azurerm_cosmosdb_account.example.name
+  throughput          = 400
+}
+
+resource "azurerm_cosmosdb_sql_container" "example" {
+  name                = "examplecontainer"
+  resource_group_name = azurerm_cosmosdb_account.example.resource_group_name
+  account_name        = azurerm_cosmosdb_account.example.name
+  database_name       = azurerm_cosmosdb_sql_database.example.name
+  partition_key_path  = "foo"
+}
+
+resource "azurerm_stream_analytics_output_cosmosdb" "example" {
+  name                      = "output-to-cosmosdb"
+  stream_analytics_job_name = azurerm_stream_analytics_job.example.name
+  resource_group_name       = azurerm_stream_analytics_job.example.resource_group_name
+  account_id                = azurerm_cosmosdb_account.example.name
+  account_key               = azurerm_cosmosdb_account.example.primary_access_key
+  database                  = azurerm_cosmosdb_sql_database.example.name
+  collection_name_pattern   = azurerm_cosmosdb_sql_container.example.id
+}
+```
+
+## Arguments Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of the Stream Output. Changing this forces a new resource to be created.
+
+* `resource_group_name` - (Required) The name of the Resource Group where the Stream Analytics Job exists. Changing this forces a new resource to be created.
+
+* `stream_analytics_job_name` - (Required) The name of the Stream Analytics Job. Changing this forces a new resource to be created.
+
+* `account_name` - (Required) The name of the CosmosDB Account.
+
+* `account_key` - (Required) The Account Key for the CosmosDB database.
+
+* `database` - (Required) The name or id of the CosmosDB database.
+
+* `collection_name_pattern` - (Required) The collection name pattern for the collection to be used. It can be the name of the container, or a collection name format which contains an optional {partition} token.
```

Comment:
As mentioned above I think it makes more sense for this to be named `container_id` - I also haven't seen an a test that provides this property with an optional partition token. Could you find an example of what that looks like? This might need to be added as an additional/separate field to the resource.